dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/MainWindow.xaml.cs

```diff
@@ -34,24 +33,25 @@ namespace VisualLayerIntegration
     /// </summary>
     public partial class MainWindow : Window
     {
-        private Random random = new Random();
-        private string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
-        private string[] customerFirstNames = new string[] { "Angel", "Josephine", "Wade", "Christie", "Whitney", "Ismael", "Alexandra", "Rhonda", "Dawn", "Roman", "Emanuel", "Evan", "Aaron", "Bill", "Margaret", "Mandy", "Carlton", "Cornelius", "Cora", "Alejandro", "Annette", "Bertha", "John", "Billy", "Randall" };
-        private string[] customerLastNames = new string[] { "Murphy", "Swanson", "Sandoval", "Moore", "Adkins", "Tucker", "Cook", "Fernandez", "Schwartz", "Sharp", "Bryant", "Gross", "Spencer", "Powers", "Hunter", "Moreno", "Baldwin", "Stewart", "Rice", "Watkins", "Hawkins", "Dean", "Howard", "Bailey", "Gill" };
+        private static readonly Random random = new Random();
+        private static readonly string chars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+        private static readonly string[] customerFirstNames = new string[] { "Angel", "Josephine",
+            "Wade", "Christie", "Whitney", "Ismael", "Alexandra", "Rhonda", "Dawn", "Roman", "Emanuel",
+            "Evan", "Aaron", "Bill", "Margaret", "Mandy", "Carlton", "Cornelius", "Cora", "Alejandro",
+            "Annette", "Bertha", "John", "Billy", "Randall" };
+        private static readonly string[] customerLastNames = new string[] { "Murphy", "Swanson", "Sandoval",
+            "Moore", "Adkins", "Tucker", "Cook", "Fernandez", "Schwartz", "Sharp", "Bryant", "Gross", "Spencer",
+            "Powers", "Hunter", "Moreno", "Baldwin", "Stewart", "Rice", "Watkins", "Hawkins", "Dean", "Howard",
+            "Bailey", "Gill" };
 
         public MainWindow()
         {
             InitializeComponent();
-        }
 
-        // Generate customers and pass data to grid.
-        private void Window_Loaded(object sender, RoutedEventArgs e)
-        {          
-            List<Customer> customers = new List<Customer>();
+            Customer[] customers = new Customer[customerFirstNames.Length];
             for (int i = 0; i < customerFirstNames.Length; i++)
             {
-                var id = new string(Enumerable.Repeat(chars, 12).Select(s => s[random.Next(s.Length)]).ToArray());
-                customers.Add(new Customer(id, customerFirstNames[i], customerLastNames[random.Next(0, customerLastNames.Length - 1)], GenerateRandomDay(), random.NextDouble() >= 0.5, GenerateRandomData()));
+                customers[i] = GenerateRandomCustomer();
             }
```

Comment:
Instead of the for loop:  
CustomerGrid.ItemsSource = customerFirstNames.Select(n => GenerateRandomCustomer()).ToArray();